common/src/main/java/com/wynntils/features/inventory/search/LevelSearchFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.inventory.search;
+
+import static java.lang.Integer.parseInt;
+
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.models.items.properties.LeveledItemProperty;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LevelSearchFilter extends WynnItemFilter {
+    private static final Pattern LEVEL_RANGE_PATTERN = Pattern.compile("^(\\d+)(-(\\d+))?$");
+
+    private int minLevel, maxLevel;
+
+    private boolean prepared = false;
+
+    public LevelSearchFilter(String searchString) {
+        super(searchString);
+    }
+
+    @Override
+    public boolean prepare() throws InvalidSyntaxException {
+        Matcher matcher = LEVEL_RANGE_PATTERN.matcher(searchString);
+        if (!matcher.find())
+            throw new InvalidSyntaxException(
+                    "feature.wynntils.containerSearch.filter.level.invalid_range", searchString);
+
+        try {
+            minLevel = parseInt(matcher.group(1));
+            maxLevel = matcher.group(3) == null ? minLevel : parseInt(matcher.group(3));
+        } catch (NumberFormatException ignore) {
+            throw new InvalidSyntaxException(
+                    "feature.wynntils.containerSearch.filter.level.invalid_range", searchString);
```

Comment:
Lot of i18n floating around the code. Perhaps you can make a custom exception, that accepts an enum you created with all the exceptions/keys.

Reply:
To be honest, I'm not quite sure to understand what you mean.
Would it be something like that ?

```
public class InvalidSyntaxException extends Exception {
    public InvalidSyntaxException(InvalidSyntaxCause cause, String... parameters) {
        super(I18n.get(cause.getTranslateKey(), parameters));
    }
    public enum InvalidSyntaxCause {
        LVL_INVALID_RANGE("feature.wynntils..."),
        PROF_INVALID_PROFESSION("feature.wynntils...");
        ...
    }
}
```